archinstall/lib/luks.py

```diff
@@ -239,13 +248,35 @@ def _add_key(self, key_file: Path):
 	def _crypttab(
 		self,
 		crypttab_path: Path,
-		key_file: Path,
-		options: List[str]
+		options: List[str],
+		key_file: Path|None = None,
+		override: bool = False
 	) -> None:
 		debug(f'Adding crypttab entry for key {key_file}')
 
-		with open(crypttab_path, 'a') as crypttab:
+		with open(crypttab_path, 'r+') as crypttab:
+			existing_data = crypttab.readlines()
 			opt = ','.join(options)
 			uuid = self._get_luks_uuid()
-			row = f"{self.mapper_name} UUID={uuid} {key_file} {opt}\n"
-			crypttab.write(row)
+			new_row = f"{self.mapper_name} UUID={uuid} {key_file or 'none'} {opt}\n"
+
+			for index, existing_row in enumerate(existing_data):
+				if uuid in existing_row:
+					if override is False:
+						print(f"Found {uuid} in {existing_row}")
```

Comment:
Can we change this  to `debug(...)` rather as it's not really relevant for the user

Reply:
Of course, it's still a draft and used print when executing blocks of code in isolated `test.py` outside of archinstall :) but good reminder!